pkg/bff/config/config.go

```diff
@@ -131,3 +145,45 @@ func (c SentryConfig) GetRelease() string {
 func (c SentryConfig) UseSentry() bool {
 	return c.DSN != ""
 }
+
+func (c AuthConfig) Validate() error {
+	if _, err := c.IssuerURL(); err != nil {
+		return err
+	}
+
+	if c.ProviderCache == 0 {
+		return errors.New("invalid configuration: auth0 provider cache duration should be longer than 0")
+	}
+	return nil
+}
+
+func (c AuthConfig) IssuerURL() (u *url.URL, err error) {
+	if c.Issuer == "" {
+		return nil, errors.New("invalid configuration: auth0 domain must be configured")
+	}
+
+	// Ensure the domain or url has a trailing slash
+	domain := c.Issuer
+	if !strings.HasSuffix(c.Issuer, "/") {
+		domain += "/"
+	}
+
+	// Attempt to parse the domain as a complete URL with a scheme.
+	if u, err = url.Parse(domain); err == nil {
+		// If a valid URL is passed in with the expected scheme and path, return it.
+		if u.Scheme != "" && u.Path == "/" && u.Host != "" {
+			return u, nil
+		}
+
+		// If an invalid URL is passed in with an unexpected scheme or path, error.
+		if u.Scheme != "" && (u.Path != "" || u.Host == "") {
+			return nil, errors.New("invalid configuration: could not parse issuer url")
+		}
+	}
+
+	// Default to the HTTPS scheme and reparse domain only configuration.
+	if u, err = url.Parse("https://" + domain); err != nil {
+		return nil, errors.New("invalid configuration: specify auth0 domain of the configured tenant")
+	}
+	return u, nil
+}
```

Comment:
This was kind of dumb - we could have just put a URL in the configuration … but I was led astray by the Auth0 documentation which used `url.Parse` to create a URL from a domain. After I implemented this and realized it wasn't necessary I expanded it to be flexible for the config to be an actual URL, just a domain, and to deal with trailing slashes (hopefully protecting the folks doing devops).